icu4c/source/common/uresbund.cpp

```diff
@@ -2055,6 +2056,47 @@ void getAllItemsWithFallback(
     }
 }
 
+struct GetAllChildrenSink : public ResourceSink {
+    // Destination sink
+    ResourceSink& dest;
+
+    GetAllChildrenSink(ResourceSink& dest)
+        : dest(dest) {}
+    virtual ~GetAllChildrenSink() override;
+    virtual void put(const char *key, ResourceValue &value, UBool isRoot,
+           UErrorCode &errorCode) override {
+        ResourceTable itemsTable = value.getTable(errorCode);
+        if (U_FAILURE(errorCode)) { return; }
+        for (int32_t i = 0; i < itemsTable.getSize(); ++i) {
+            itemsTable.getKeyAndValue(i, key, value);
+            if (value.getType() == URES_ALIAS) {
+                ResourceDataValue& rdv = static_cast<ResourceDataValue&>(value);
+                StackUResourceBundle stackTempBundle;
+                UResourceBundle* aliasRB = getAliasTargetAsResourceBundle(rdv.getData(), rdv.getRes(), key, -1, rdv.getValidLocaleDataEntry(), nullptr, 0, stackTempBundle.getAlias(), &errorCode);
+                if (U_SUCCESS(errorCode)) {
+                    ResourceDataValue aliasedValue;
+                    aliasedValue.setData(aliasRB->getResData());
+                    aliasedValue.setValidLocaleDataEntry(aliasRB->fValidLocaleDataEntry);
+                    aliasedValue.setResource(aliasRB->fRes, ResourceTracer(aliasRB));
+                    dest.put(key, aliasedValue, isRoot, errorCode);
+                }
+            } else {
+                dest.put(key, value, isRoot, errorCode);
+            }
```

Comment:
after this line please
```
if (U_FAILURE(errorCode)) { return; }
```

Reply:
I'm making this change, but I have reservations.  I'm not sure the right answer to an error following an alias (or passing a value to the resource sink) should be to abort the whole iteration.

If the sink fails to accept the value, maybe that's a good reason to quit passing it stuff (the failure check wasn't in there before, but because we also don't clear the error code, we were getting that effect much more expensively, so maybe that's grounds for adding the failure check).

On the other hand, if we fail to follow the alias, I'd argue it's better to throw out that one value and keep iterating.  I think your argument was that we're getting the "wrong answer" in their case, and so it'd be better to be spectacularly wrong so that we have more of a chance of detecting that there's an error at all, since the error would be in our own resource bundles.  I understand the rationale here, but in real application code, it seems like the "do the best you can" approach would be less destructive.

As I said, I'm making the change anyway, but I'm not crazy about it.

Reply:
See my earlier reply (https://github.com/unicode-org/icu/pull/1777#discussion_r695280415) about UErrorCode emulating exceptions. If there was a good case for why we would catch and ignore an exception, then we would have a case for ignoring and resetting a failure code.

In this case it looks like you are right too, and the current version of the code actually merely becomes an *inefficient* no-op without the additional line.